cmd/nerdctl/run.go

```diff
@@ -737,6 +740,79 @@ func createContainer(ctx context.Context, cmd *cobra.Command, client *containerd
 	return container, nil, nil
 }
 
+// TODO This function is copied from /cmd/pull, need to be refactored.
```

Comment:
what refactor to be needed here? E.g., why can't we import `pkg/cmd/image` and use `image.EnsureImage` directly here?

Reply:
I have rewritten this function. It should look a bit better.